apps/glusterfs/app_node.go

```diff
@@ -320,6 +321,49 @@ func (a *App) NodeDelete(w http.ResponseWriter, r *http.Request) {
 				return err
 			}
 
+			volumes, err := ListCompleteVolumes(tx)
+			if err != nil {
+				logger.Err(err)
+				return err
+			}
+			for _, volID := range volumes {
+				volEntry, err := NewVolumeEntryFromId(tx, volID)
+				if err != nil {
+					logger.LogError("Get volume entry for ID %s Failed with error %s", volID, err.Error())
+					continue
+				}
+				if volEntry.Info.Block {
+					for _, id := range volEntry.Info.BlockInfo.BlockVolumes {
+
+						blockEntry, err := NewBlockVolumeEntryFromId(tx, id)
+						if err != nil {
+							logger.LogError("Get block volume entry for ID %s Failed with error %s", id, err.Error())
+							continue
+						}
+
+						if err = blockEntry.updateHosts(txdb); err != nil {
```

Comment:
This strikes me as rather suboptimal. We're in a nested loop calling the same "db queries" over and over again in a transaction such that we're always going to get the same results. It strikes me as more correct to simply fetch the list of nodes for the cluster and re-use that list throughout the loop. See my comment on `updateMountInfo` for more details.